Import `progress-events-generated-correctly.html` from Blink / Chromium

Ahmad-S792 opened the pull request:
#### a18f3e115c5277f715d33631a60462e361bd2dc4
<pre>
Import `progress-events-generated-correctly.html` from Blink / Chromium

<a href="https://bugs.webkit.org/show_bug.cgi?id=269644">https://bugs.webkit.org/show_bug.cgi?id=269644</a>

Reviewed by Eric Carlson.

This patch is to import test from Blink, which was updated to use `testharness`.

Partial Merge: <a href="https://source.chromium.org/chromium/chromium/src/+/25f67c81bfa8c6b1ce5a4c061ca5bd76eae32fff">https://source.chromium.org/chromium/chromium/src/+/25f67c81bfa8c6b1ce5a4c061ca5bd76eae32fff</a>

* LayoutTests/media/progress-events-generated-correctly.html: Remove old copy
* LayoutTests/media/progress-events-generated-correctly-expected.txt: Ditto
* LayoutTests/http/tests/media/progress-events-generated-correctly.html: Updated Test Case in `http` directory
* LayoutTests/http/tests/media/progress-events-generated-correctly-expected.txt: Add Test Case Expectation

&gt; Remove / Update old Test Expectations:
* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac/TestExpectations:

Canonical link: <a href="https://commits.webkit.org/275082@main">https://commits.webkit.org/275082@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3b9b03cbd5dc295fcea92182c4ce3889ea8c58e8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/40392 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/19404 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/42770 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/42938 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/36481 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/22351 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/16733 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/33516 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/40966 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/16356 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/34822 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/14086 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/14150 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/35787 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/44215 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/36617 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/36110 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/39864 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/15207 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/12452 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/38145 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/16826 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/35070 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9161 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/16875 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/16470 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

jyavenard commented on LayoutTests/platform/mac/TestExpectations:
```diff
@@ -746,7 +746,7 @@ webkit.org/b/136708 media/media-fullscreen-not-in-document.html
 
 # This test requires generation of progress events during loading
-webkit.org/b/100984 media/progress-events-generated-correctly.html [ Failure ]
+webkit.org/b/100984 http/tests/media/progress-events-generated-correctly.html [ Failure ]
```
why is it a failure on Mac and not iOS ?

Ahmad-S792 replied:
Because of this: This test requires generation of progress events during loading

Ahmad-S792 replied:
> why is it a failure on Mac and not iOS ?

and due to this bug. https://bugs.webkit.org/show_bug.cgi?id=101439

Ahmad-S792 replied:
So I created new bug to at least update test (modernise). It was still failing on macOS but iOS and others were passing.

Ahmad-S792 replied:
@jyavenard - If you are OK with above answers, I will merge this, I stopped earlier based on your query.

Ahmad-S792 replied:
@jyavenard Should I go ahead and merge it? It might need to be rebased but wanted to get your input based on your question.

Ahmad-S792 committed:
Import `progress-events-generated-correctly.html` from Blink / Chromium

https://bugs.webkit.org/show_bug.cgi?id=269644

Reviewed by Eric Carlson.

This patch is to import test from Blink, which was updated to use `testharness`.

Partial Merge: https://source.chromium.org/chromium/chromium/src/+/25f67c81bfa8c6b1ce5a4c061ca5bd76eae32fff

* LayoutTests/media/progress-events-generated-correctly.html: Remove old copy
* LayoutTests/media/progress-events-generated-correctly-expected.txt: Ditto
* LayoutTests/http/tests/media/progress-events-generated-correctly.html: Updated Test Case in `http` directory
* LayoutTests/http/tests/media/progress-events-generated-correctly-expected.txt: Add Test Case Expectation

> Remove / Update old Test Expectations:
* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac/TestExpectations:

Canonical link: https://commits.webkit.org/275082@main
